### PR TITLE
feat: basic gui classes

### LINKED ISSUE #8: Create a ToolBox with ImGUI
Create a some GUI window to see how dear IMGUI work

### LINKED ISSUE #9: Abstract ImGui Window and Widget
Create an abstraction of ImGui window and widgets for convenience and usability.
TODO:
 - Create an abstract window class
 - Create abstract Widgets class

### PR DESCRIPTION
**Pull request by Perzi-Lgs**
Create basic classes of Gui

### CODE REVIEW

**Review by Mathieu-Lala — APPROVED** [review]
Could use std::remove_if and std::find_if sometimes

**Review by Mathieu-Lala — APPROVED** [review]
Could use std::remove_if and std::find_if sometimes

File: src/Engine/include/Engine/UI/AWidget.hpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#include <functional>
+
+namespace usa {
+
+namespace Engine {
+
+class AWidget
+{
+protected:
+    std::string m_name{};
```
Comment (rmaxi-me) [inline src/Engine/include/Engine/UI/AWidget.hpp]:
```suggestion
    std::string_view m_name{};
```

File: src/Engine/include/Engine/UI/UiWindow.hpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    class UiWindow
+    {
+    private:
+        ImGuiWindowFlags m_window_flags{0};
+        bool m_isOpen = true;
+
+        ImVec2 m_size{};
+        ImVec2 m_position{};
+    protected:
+        std::string m_name{};
```
Comment (rmaxi-me) [inline src/Engine/include/Engine/UI/UiWindow.hpp]:
```suggestion
        std::string_view m_name{};
```

File: src/Engine/include/Engine/UI/Widget/WMenuItem.hpp [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+namespace usa
+{
+
+    namespace Engine
+    {
+
+        class WMenuItem : public AWidget
+        {
+        private:
+            std::string m_shortcut{};
```
Comment (rmaxi-me) [inline src/Engine/include/Engine/UI/Widget/WMenuItem.hpp]:
```suggestion
            std::string_view m_shortcut{};
```

File: src/Engine/include/Engine/UI/Widget/WSeparator.hpp [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    namespace Engine
+    {
+
+        class WSeparator : public AWidget
+        {
+        private:
+            std::function<void()> f;
+
+        public:
+            WSeparator(std::string name, ImVec2 size = {}) : AWidget(name, size) {}
```
Comment (rmaxi-me) [inline src/Engine/include/Engine/UI/Widget/WSeparator.hpp]:
```suggestion
            WSeparator(const std::string_view &name, ImVec2 size = {}) : AWidget(name, size) {}
```

File: src/Engine/src/Application.cpp [hunk 5]
```diff
@@ -52,8 +52,10 @@ auto usa::Engine::Application::start(const std::string_view &title) -> void
 {
     sf::Event event{};
     sf::Uint32 style = sf::Style::Close;
+    sf::Uint32 fullscreen = sf::Style::Fullscreen;
+    sf::Uint32 resize = sf::Style::Resize;
 
-    style |= m_settings.fullscreen ? sf::Style::Fullscreen : sf::Style::Resize;
+    style |= m_settings.fullscreen ? fullscreen : resize;
```
Comment (rmaxi-me) [inline src/Engine/src/Application.cpp:58]: Do we really need 2 variables for this?

File: src/EpiGimp/src/imgui_demo.cpp [hunk 6]
```diff
@@ -0,0 +1,5266 @@
+// dear imgui, v1.77 WIP
```
Comment (rmaxi-me) [inline src/EpiGimp/src/imgui_demo.cpp]: Unused file

File: src/EpiGimp/src/EpiGimpApp.cpp [hunk 7]
```diff
@@ -27,21 +37,31 @@ auto usa::EpiGimpApp::init() -> void { createScene<SceneCanvas>(800, 800); }
...
 auto usa::EpiGimpApp::draw() -> void
 {
-    ImGui::Begin("Test window");
-    {
-        ImGui::Text("Hello world!");
-        ImGui::Button("> Useless button <");
-    }
-    ImGui::End();
+    static bool cpt = false;
+    static Engine::UiWindow *win = new Engine::MainMenuBar();
```
Comment (rmaxi-me) [inline src/EpiGimp/src/EpiGimpApp.cpp]: Leaks

File: src/EpiGimp/src/EpiGimpApp.cpp [hunk 8]
```diff
@@ -27,21 +37,31 @@ auto usa::EpiGimpApp::init() -> void { createScene<SceneCanvas>(800, 800); }
...
-        ImGui::ColorPicker4("Color picker", color);
-        ImGui::Button("> Useless button <");
+        Engine::Menu menu("Fichiers");
+        std::shared_ptr<Engine::UiWindow> Mmenu = std::make_shared<Engine::Menu>(menu);
+
+        auto i1 = std::make_shared<Engine::WMenuItem>("File", "CTRL+N");
+        auto i2 = std::make_shared<Engine::WMenuItem>("Create");
+        auto i3 = std::make_shared<Engine::WMenuItem>("Open", "CTRL+O");
+        auto i4 = std::make_shared<Engine::WMenuItem>("Open as Layers", "CTRL+ALT+O");
+        auto i5 = std::make_shared<Engine::WMenuItem>("Open Location");
```
Comment (rmaxi-me) [inline src/EpiGimp/src/EpiGimpApp.cpp]: Do we really need dynamic allocation?

**Review by rmaxi-me — CHANGES_REQUESTED** [review]

File: src/Engine/src/Application.cpp (hunk 5 above)
Comment (Perzi-Lgs) [reply src/Engine/src/Application.cpp:58]: That wasn't compiling on my computer cause of cast of different size

**Review by rmaxi-me — APPROVED** [review]

**Pull request merged**